dgraph/cmd/graphql/schema/schema_test.go

```diff
@@ -134,56 +82,65 @@ func TestEqualSchema(t *testing.T) {
 		str1, err := ioutil.ReadFile(fileName1)
 		if err != nil {
 			t.Errorf("Unable to read file %s", fileName1)
-		}
-
-		doc1, gqlerr := parser.ParseSchema(&ast.Source{Input: string(str1)})
-		if gqlerr != nil {
-			t.Errorf("Unable to parse file %s\n%s", fileName1, gqlerr.Error())
 			continue
 		}
 
-		gqlErrList1 := ValidateSchema(doc1)
-		if gqlErrList1 != nil {
-			t.Errorf("Unable to validate schemafile %s\n%s", fileName1, gqlErrList1.Error())
-		}
-
-		AddScalars(doc1)
-
-		schema1, gqlerr := validator.ValidateSchemaDocument(doc1)
-		if gqlerr != nil {
-			t.Errorf("Unable to validate schemafile %s\n%s", fileName1, gqlerr.Error())
+		sch1, errlist1 := GenerateCompleteSchema(string(str1))
```

Comment:
This test can ideally be merged with the `TestSchemaString` function using Table Driven Tests above or are they testing different things?